src/features/dashboard/Chart/panZoomPlugin.ts

```diff
@@ -277,7 +277,16 @@ export default {
             }
 
             const { data, options } = getState(chart);
-            options.maxRange = data[data.length - 1].timestamp;
+            if (data.length === 0) {
+                return;
+            }
```

Comment:
No need for this check because data.at(-1) will return undefined if index is `< -length` or `>= length`

Reply:
I was a bit unsure if I wanted to leave it or not, and I decided to leave it because it gives a bit more explicit meaning to why you need check for it. I think when this was implemented, we did not really consider this corner case at all, and leaving this in will remind us to check for similar edge-cases in the future.

I do agree, though, that this is a bit redundant. But still, why not 🤷 🤓